File: android/src/main/kotlin/np/com/kedark/html_to_image/CaptureStrategy.kt
package np.com.kedark.html_to_image

import android.util.DisplayMetrics

class CaptureStrategy(
    val width: Int?,
    val height: Int?,
    val script: String?
) {
    companion object {
        fun parseFromMap(map: Map<*, *>, displayMetrics: DisplayMetrics): CaptureStrategy {
            val width =
                (map["width"] as? Int?)?.let { it * displayMetrics.density }
            val height = (map["height"] as? Int?)?.let { it * displayMetrics.density }
            val script = map["script"] as? String
            return CaptureStrategy(width?.toInt(), height?.toInt(), script)
        }
    }
}

File: example/ios/.symlinks/plugins/html_to_image/ios/Classes/CaptureStrategy.swift
class CaptureStrategy {
    let width: Int?
    let height: Int?
    let script: String?

    init(width: Int?, height: Int?, script: String?) {
        self.width = width
        self.height = height
        self.script = script
    }

    static func parseFromMap(_ map: [String: Any]) -> CaptureStrategy {
        let width = map["width"] as? Int
        let height = map["height"] as? Int
        let script = map["script"] as? String
        return CaptureStrategy(width: width, height: height, script: script)
    }
}